src/services/WebSocketService.js

```diff
@@ -0,0 +1,121 @@
+import url from '/util/url'
+
+class WebSocketService {
```

Comment:
I'm generally antagonistic towards classes (in JavaScript). I touch on it [here](https://gist.github.com/staydecent/0a6588cc1068f3bab394abfd3add02e2).

If you do need various instances of an object, the you can use [factory functions](https://medium.com/javascript-scene/javascript-factory-functions-with-es6-4d224591a8b1).

In this case, it looks like it's a singleton? If so, that can be replaced with a module-level reference:

```javascript
let cachedInstance

// ... later in your code
const instance = cachedInstance || createInstance()
```

Moreover, it looks like you wrap the usage of this inside a useEffect call. Why not make a custom hook instead of the singleton class? I think `util` is a good enough folder for custom hooks.

Something like `useWebsocket`. Here are a few examples of custom hooks:

- https://gist.github.com/staydecent/d8f9ab4f34347e87f87379da819e928a
- https://github.com/inputlogic/elements/blob/master/components/use-successive-taps/index.js
- https://github.com/inputlogic/elements/blob/master/components/use-mapped-state/index.js

And the React Docs: https://reactjs.org/docs/hooks-custom.html

Let me know if I can help out more with this!

Reply:
New use-websocket hook has been added in the hooks folder, just working out the details now.